core/src/test/java/org/wso2/msf4j/ResponseTest.java

```diff
@@ -0,0 +1,110 @@
+import org.wso2.carbon.messaging.CarbonMessage;
+import org.wso2.carbon.messaging.DefaultCarbonMessage;
+import org.wso2.carbon.messaging.Headers;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.AssertJUnit.assertFalse;
+import static org.testng.AssertJUnit.assertTrue;
+
+public class ResponseTest {
+
+    private Response response;
+
+    @BeforeClass
+    public void setup() {
+        response = new Response(CarbonMessage::release);
+    }
+
+    @Test
+    public void testIsEomAddedSet() {
+        assertFalse(response.isEomAdded());
+    }
+
+    @Test
+    public void testIsEmpty() {
+        assertTrue(response.isEmpty());
+    }
+
+    @Test
+    public void testHeaders() {
+        String key = "Client";
+        String value = "WSO2";
+
+        Headers headers = response.getHeaders();
+        assertTrue(headers.size() == 0);
+
+        response.setHeader(key, value);
+        assertTrue(response.getHeaders().size() == 1);
+        assertEquals(response.getHeader(key), value);
+
+        String header1Key = "Browser";
+        String header1Val = "Chrome";
+        String header2Key = "Region";
+        String header2Val = "US";
+        String header3Key = "Secure";
+        String header3Val = "true";
+        Map<String, String> customHeaders = new HashMap<>();
+        customHeaders.put(header1Key, header1Val);
+        customHeaders.put(header2Key, header2Val);
+        customHeaders.put(header3Key, header3Val);
+        response.setHeaders(customHeaders);
+        assertTrue(response.getHeaders().size() == 4);
+        assertEquals(response.getHeader(header1Key), header1Val);
+        assertEquals(response.getHeader(header2Key), header2Val);
+        assertEquals(response.getHeader(header3Key), header3Val);
+
+        response.removeHeader(header1Key);
+        assertNull(response.getHeader(header1Key));
+        assertTrue(response.getHeaders().size() == 3);
+    }
+
+    @Test
+    public void testProperties() {
+        String key = "key";
+        Object value = new Request(new DefaultCarbonMessage());
+
+        Map<String, Object> properties = response.getProperties();
+        assertTrue(properties.size() == 0);
+
+        response.setProperty(key, value);
+        assertTrue(response.getProperties().size() == 1);
+        assertEquals(response.getProperty(key), value);
+
+        response.removeProperty(key);
+        assertTrue(response.getProperties().size() == 0);
+        assertNull(response.getProperty(key));
+    }
+
+    @Test
+    public void testCarbonMessage() {
+        CarbonMessage carbonMessage = response.getCarbonMessage();
+        assertTrue(carbonMessage.isEmpty());
+        assertTrue(carbonMessage.isBufferContent());
+        assertFalse(carbonMessage.isEndOfMsgAdded());
+        assertFalse(carbonMessage.isAlreadyRead());
+        assertFalse(carbonMessage.isFaulty());
+    }
+}
```

Comment:
We need a line break at the end of the file.

Reply:
Line break is already there at the end of the file